static/app/views/starfish/views/spans/useModuleSort.ts

```diff
@@ -0,0 +1,44 @@
+import {fromSorts} from 'sentry/utils/discover/eventView';
+import type {Sort} from 'sentry/utils/discover/fields';
+import {useLocation} from 'sentry/utils/useLocation';
+import {SpanMetricsFields, StarfishFunctions} from 'sentry/views/starfish/types';
+import {QueryParameterNames} from 'sentry/views/starfish/views/queryParameters';
+
+type Query = {
+  [QueryParameterNames.SORT]: string;
+};
+
+const SORTABLE_FIELDS = [
+  `avg(${SpanMetricsFields.SPAN_SELF_TIME})`,
+  `${StarfishFunctions.HTTP_ERROR_COUNT}()`,
+  `${StarfishFunctions.SPM}()`,
+  `${StarfishFunctions.TIME_SPENT_PERCENTAGE}()`,
+  `${StarfishFunctions.TIME_SPENT_PERCENTAGE}(local)`,
+] as const;
```

Comment:
m: Why an array over a set like we had before?

if it's for the type, let's just make a string enum and then make a set from the string enum.

Something like

```ts
enum SortableFields {
  ...
}

const DEFAULT_SORTABLE_FIELDS = new Set(Object.values(SortableFields));
```

Reply:
Yep, it's for the type. Using a plain array makes it pretty easy to generate a type _and_ check for inclusion. Is the enum approach better in some way?

Reply:
I think the enum allows for the individual fields to be used easier. For ex when we have to construct DEFAULT_SORT below:

```
field: SortableFields.TimeSpent,
```

it also makes it easier to understand the intent of the `SORTABLE_FIELDS` a lot easier. I'm fine to keep it as array though since I understand that is personal preference.

Reply:
Ahh yes yes! I take your point! I'm going to leave this as-is for now, but I'm working on some typing improvements to the entirely of Starfish span views that I hope will cover this 👌🏻